File: common/src/main/java/io/xol/chunkstories/world/WorldLoadingException.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.world;

public class WorldLoadingException extends Exception {

	private static final long serialVersionUID = -7131921980416653390L;

	public WorldLoadingException(String message, Throwable cause) {
		super(message, cause);
	}

	@Override
	public String getMessage() {
		if(getCause() != null)
			return super.getMessage()+": "+getCause().getClass().getSimpleName()+"\n"+getCause().getMessage();
		return super.getMessage();
	}

	public WorldLoadingException(String message) {
		super(message);
	}

}

File: common/src/main/java/xyz/chunkstories/world/WorldLoadingException.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.world

class WorldLoadingException : Exception {

    constructor(message: String, cause: Throwable) : super(message, cause) {}

    override val message: String?
        get() {
            return if (cause != null) super.message + ": " + cause.javaClass.simpleName + "\n" + cause.message else super.message
        }

    constructor(message: String) : super(message) {}

    companion object {

        private val serialVersionUID = -7131921980416653390L
    }

}
